### PR TITLE
Updates for ARC feedback #182, #198 and #201

### PR DESCRIPTION
**Pull request by vlsunil**
My attempt to resolve ARC feedback regarding UEFI variable persistence, SATP and ACPI device property section.  Please review and suggest better wordings or if I am missing any thing.

### CODE REVIEW

**Comment by vlsunil** [comment]
@andreiw @jhauser-us @avpatel @adurbin-rivos - please review.

**Review by andreiw — COMMENTED** [review]
The intention of URT_040 was to avoid RPi-like UEFI implementations, where NV variables are in practice volatile after ExitBootServices. This leads to bad experiences with OS installers and probably other things.

Pi 4 had no non-volatile store beyond the SD card. Variables were persisted in RPI_EFI.FD during Boot Services, but he SD card could not be accessed after OS use, so - no non-volatile vars in the OS.

URT_040 refers to a behavior wrt ResetSystem, and that was intended - again, deficient designs like RPi4 could, during their ResetSystem path, flush out NV variables to persistent store (now that the OS is out of the way)

**Review by andreiw — COMMENTED** [review]
I hate to be pendantic, but the UEFI spec *already* talks about identity mapping for RAM regions as a requirement.

Probably want an additional requirement that EFI_MEMORY_ATTRIBUTE_PROTOCOL and EFI_CPU_ARCH_PROTOCOL.SetMemoryAttributes() are implemented and non-dummy (really affects page attributes).

**Comment by adurbin-rivos** [comment]
It seems that people feel that system designs should follow the already written requirements in the specs vs trying to re-describe or over-describe things to keep people from shooting themselves in the foot. I'm fine w/ these changes as I'm more on the side of "people should design things correctly" vs trying to prevent larger mistakes.

**Review by adurbin-rivos — APPROVED** [review]

**Comment by ved-rivos** [comment]
If its a common mistake then its not a bad idea to call it out. The rule book is intended to be helpful.

File: uefi.adoc [hunk 1]
```diff
@@ -19,8 +19,9 @@ IMPORTANT: All content in this section is optional and recommended for BRS-B.
 | `UEFI_030` a| MUST meet the following memory map rules:
 
               * The default memory space attribute is `EFI_MEMORY_WB`.
-              * Address translation is enabled.
+              * Paged virtual-memory scheme MUST be configured with identity mapping to support memory protection.
               * Only use `EfiRuntimeServicesData` memory type for describing any SMBIOS data structures.
+2+| _Enabling paged virtual memory scheme means satp.MODE must not be set to `Bare` and proper page table for identity mapping is setup by the firmware. This is required for platform protection use cases before handing off to the OS._
```
Comment (avpatel) [inline uefi.adoc:24]: s/proper page table for identity mapping/proper page table with identity mapping/

I think the page table is setup by OS and this page table identity mapping is required when OS makes EFI call at boot-time or run-time.

**Comment by vlsunil** [comment]
> The intention of URT_040 was to avoid RPi-like UEFI implementations, where NV variables are in practice volatile after ExitBootServices. This leads to bad experiences with OS installers and probably other things.
> 
> Pi 4 had no non-volatile store beyond the SD card. Variables were persisted in RPI_EFI.FD during Boot Services, but he SD card could not be accessed after OS use, so - no non-volatile vars in the OS.
> 
> URT_040 refers to a behavior wrt ResetSystem, and that was intended - again, deficient designs like RPi4 could, during their ResetSystem path, flush out NV variables to persistent store (now that the OS is out of the way)

Ahh OK. Thanks!. I thought that was the intention. In that case, let's rephrase the requirement that "The UEFI variables MUST be persistent across ResetSystem() call".  I hope that should be clear enough requirement.

**Comment by vlsunil** [comment]
> I hate to be pendantic, but the UEFI spec _already_ talks about identity mapping for RAM regions as a requirement.
>
While I agree with that, I think it is better to mention again here since we are talking about MMU configuration. This should make it clear that we are still requiring identity map while configuring non-BARE mode.

> Probably want an additional requirement that EFI_MEMORY_ATTRIBUTE_PROTOCOL and EFI_CPU_ARCH_PROTOCOL.SetMemoryAttributes() are implemented and non-dummy (really affects page attributes).

Good point. However, it is better to avoid requiring EFI_CPU_ARCH_PROTOCOL which is part of PI spec. Better to leave to the FW implementation to provide EFI_MEMORY_ATTRIBUTE_PROTOCOL even when they are not PI complaint. WDYT?

File: uefi.adoc (hunk 1 above)
Comment (vlsunil) [reply uefi.adoc:24]: > s/proper page table for identity mapping/proper page table with identity mapping/
> 
Thanks!. Let me update.

**Comment by xypron** [comment]
> In that case, let's rephrase the requirement that "The UEFI variables MUST be persistent across ResetSystem() call". I hope that should be clear enough requirement.

This get's it wrong. Volatile variables should not be persisted. Only variables with attribute EFI_VARIABLE_NON_VOLATILE must be persisted.

**Pull request merged**